terraform/aws/modules/cluster/worker_asg.tf

```diff
@@ -83,3 +25,68 @@ data:
         - system:nodes
 CONFIGMAPAWSAUTH
 }
+
+
+resource "aws_launch_template" "cluster_nodes_eks_launch_template" {
+  name        = "cnc_cluster_launch_template"
+  description = "CnC cluster nodes launch template"
+
+  vpc_security_group_ids = [aws_security_group.worker-sg.id]
+
+  block_device_mappings {
+    device_name = "/dev/xvda"
+
+    ebs {
+      volume_size = 20
+      volume_type = "gp2"
+    }
+  }
+
+  image_id      = var.eks_ami_id
+  instance_type = var.instance_type
+
+  user_data = base64encode(local.worker-userdata)
+
+  tag_specifications {
+    resource_type = "instance"
+
+    tags = {
+      Name = "cnc-cluster-nodes"
+    }
+  }
+}
+
+resource "aws_eks_node_group" "general_nodes_eks_cluster_ng" {
+  cluster_name    = aws_eks_cluster.cluster.name
+  node_group_name = "general-nodes"
```

Comment:
do we want to give a dynamic map from a variable and prepend or append `-nodes` or any other convention we want